### PR TITLE
feat(widget-builder): Add visualize field

### PR DESCRIPTION
**Pull request by narsaynorath**
Adds a foundation for the visualize field. This field is pretty complex so I've had to break it down into smaller tasks as I discovered them. For this PR it's focused on charts and aggregates, so tables aren't guaranteed to work and it'll only allow you to select functions right now.

Functionality involved (assuming timeseries chart selected):
- the argument/column selector and the aggregate selectors should show the right values for the dataset
  - Note: Going between datasets doesn't automatically flush the existing values, so just look at the values that get presented in the dropdowns and verify they're the same as the old widget builder
  - I noticed there's an extra `span.duration` in the release dataset, I'll follow up about it, but ignore that for now
- You can add a series
- You can delete a series iff there's more than 1 series selected
- an "Add Equation" button is shown but it doesn't do anything right now
- Custom measurements should appear in the transactions dataset

Sets the foundation for #81506

### CODE REVIEW

File: static/app/views/dashboards/widgetBuilder/components/visualize.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+  // Used to extract selected aggregates and parameters from the fields
+  const stringFields = fields?.map(generateFieldAsString);
+
+  return (
+    <Fragment>
+      <SectionHeader
+        title={t('Visualize')}
+        tooltipText={t(
+          "This is the data you'd be visualizing in the display. If the overlay units conflict, the charts will always base it off of the first line."
```
Comment (nikkikapadia) [inline static/app/views/dashboards/widgetBuilder/components/visualize.tsx:76]: just a heads up, when I was talking to Vasudha about the tooltip texts she said we're taking some from explore and Visualize is one of them. She told me to use this when I asked her 

image

File: static/app/views/dashboards/widgetBuilder/components/visualize.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+                    // TODO: Handle scalars (i.e. no aggregate, for tables)
+                    // Update the current field's aggregate with the new aggregate
+                    if (field.kind === 'function') {
+                      field.function[1] = newField.value as string;
+                    }
+                    dispatch({
+                      type: updateAction,
+                      payload: fields,
+                    });
+                  }}
```
Comment (nikkikapadia) [inline static/app/views/dashboards/widgetBuilder/components/visualize.tsx:123]: is there going to be a way to clear this field or select none after you've selected something? I noticed that there's aggregate functions that don't take any columns so i'm just wondering 👀

**Review by nikkikapadia — COMMENTED** [review]
Looking good just a couple comments

File: static/app/views/dashboards/widgetBuilder/components/visualize.tsx (hunk 2 above)
Comment (narsaynorath) [reply static/app/views/dashboards/widgetBuilder/components/visualize.tsx:123]: Yeah I'm going to follow up on how these different fields work. The designs haven't really taken into account all of the "different" kind of aggregates, e.g. we have aggregates like `count_if` that take 3 args.

We could possibly allow clearing the column selector, that would be aligned with how a single field can be selected in the table (we're going to allow selecting `None` in the aggregates). I thought it'd be intuitive to make it a primary option in the column selector, and special case it so it doesn't show the aggregate selection where it doesn't make sense, but we'll see what works.

Comment (nikkikapadia) [reply static/app/views/dashboards/widgetBuilder/components/visualize.tsx:123]: ok ya sounds good! i'll slap the approve

**Review by nikkikapadia — APPROVED** [review]

**Pull request merged**